### PR TITLE
Feature/list of set goals

### PR DESCRIPTION
**Pull request by AliaksandraNovikava**
- Created the component NewGoalPreview to render a preview of a new goal
- Styled the preview
- Created the component NewGoalsList to render a list of all previews
- Styled the list
- Installed and implemented use-local-storage-state hook

### CODE REVIEW

**Review by doemser — CHANGES_REQUESTED** [review]
At first glance your code looks good, however when I tried your app it broke:

When I have a modal to add a goal open and click outside to close it, an error appears and I have to clear the localStorage to get the app working again. This needs to be fixed.

Despite that I <3 the feeling of your app and your coding feels proper and solid. Really good work so far.

**Comment by AliaksandraNovikava** [comment]
Thanks for your review @doemser 
I didn't notice this problem at all😅

**Comment by doemser** [comment]
> Thanks for your review @doemser I didn't notice this problem at all😅

I am really not sure about the solution you came up with, there you are directly talking to the localStorage Browser API, while you also use the use-local-storage-state library.

I would throw out everything where you address localStorage directly, you can rely on the use-local-storage-state hook, only with the setFunction from it you should handle states in local storage

**Review by vaupunkt — APPROVED** [review]
very impressive! 👀🎉

I really like it. one thing: will there be a option to see how many repetitions I chose for myGoals? :) 
maybe thats something you are going to add in the future anyway 😃

**Comment by AliaksandraNovikava** [comment]
@doemser My code works just fine even without using the useEffect hook to store new data in the browser localStorage 👀 I think I've made things unnecessarily complicated. Hope that my PR is now ready to be approved🙏

File: components/NewGoalsList/index.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          <StyledCard key={goal.id}>
+            <NewGoalPreview
+              image={goal.icon}
+              title={goal.name}
+              creationDate={goal.creationDate}
+            />
+          </StyledCard>
+        ))}
+      </StyledList>
+    </>
```
Comment (doemser) [inline components/NewGoalsList/index.js]: you do not need this fragments

**Comment by AliaksandraNovikava** [comment]
@vaupunkt The user will be able to see more information on each goal in a goals detail view which I will cover in a later user story.

File: pages/index.js [hunk 2]
```diff
@@ -91,7 +99,10 @@ export default function HomePage({ goals, categoryColors }) {
 
   useEffect(() => {
     const handleEscapeKeyPress = (e) => {
-      if (e.key === "Escape" && isModalOpen) {
+      if (
+        e.key === "Escape" &&
+        localStorage.getItem("isModalOpen") === "true"
```
Comment (doemser) [inline pages/index.js]: here you are still using the localStorage API directly

```suggestion
       isModalOpen
```

**Review by doemser — CHANGES_REQUESTED** [review]
Almost done on this PR! :D

**Comment by MariaRiosNavarro** [comment]
LGTM🚀 Everything was done

**Comment by AliaksandraNovikava** [comment]
@MariaRiosNavarro thanks for your approval👍

**Pull request merged**